kubernetes/katago-server/templates/celery-flower/deployment.yaml

```diff
@@ -0,0 +1,79 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: celery-flower-deployment
+  namespace: serverspace
+  labels:
+    app: celery-flower
+    {{- include "katago-server.labels" . | nindent 4 }}
+spec:
+  replicas: {{.Values.celeryFlower.replicaCount}}
+  selector:
+    matchLabels:
+      app: celery-flower
+      {{- include "katago-server.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        app: celery-flower
+        {{- include "katago-server.selectorLabels" . | nindent 8 }}
+      annotations:
+        checksum/django_env_secrets: {{ include (print $.Template.BasePath "/env/django_env_secrets.yaml") . | sha256sum }}
```

Comment:
How does that work ? does it require to modify the template ?

Reply:
I am simply doing the recommendation that *you* linked! Right?
https://github.com/katago/katago-server/pull/44#discussion_r517000159

My understanding of workflow was that we will have a secrets file env/django_env_secrets.yaml sitting on disk (unencrypted while it is on disk?) that we deploy, the same way we have a secrets file in docker compose sitting (unencrypted) on disk that is mounted by production.yml into the environment variables for a container.

This uses the mechanism in the link you have - if the secrets file changes, then the sha256sum will change, and so this annotation will change, and then helm will know that this deployment needs to be updated and to do a rolling restart properly.

Reply:
Is there a different workflow you expected?

Reply:
makes sense.

I wasn't planing to deploy secrets with helm so I was thinking of only doing checksum for configmap (that can be deploy with helm, same as https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments).

Also based on

![image](https://user-images.githubusercontent.com/13785185/98447376-b4c0b700-2124-11eb-8d69-a2e0b5b8cba5.png)

I'm not sure we need it for env variables (sorry if I made you do useless stuff).